lambdas/InspectMediaLambdaFunction/audio.js

```diff
@@ -0,0 +1,95 @@
+const ebur128 = require('./ebu-r-128');
+const ffprobe = require('./ffprobe');
+const mpck = require('./mpck');
+
+/** @typedef {import('./index').InspectTask} InspectTask */
+
+/**
+ * @typedef {object} AudioInspection
+ * @property {number} [Duration]
+ * @property {string} [Format]
+ * @property {number} [Bitrate]
+ * @property {number} [Frequency]
+ * @property {number} [Channels]
+ * @property {string} [Layout]
+ * @property {string} [Layer]
+ * @property {number} [Samples]
+ * @property {number} [Frames]
+ * @property {number} [LoudnessIntegrated]
+ * @property {number} [LoudnessTruePeak]
+ * @property {number} [LoudnessRange]
+ */
+
+module.exports = {
+  /**
+   * @param {InspectTask} task
+   * @param {string} filePath
+   * @returns {Promise<AudioInspection>}
+   */
+  inspect: async function inspect(task, filePath) {
+    /** @type {AudioInspection} */
+    const inspection = {};
+
+    // FFprobe data
+    try {
+      const probe = await ffprobe.inspect(filePath);
+      const stream = probe.streams.find((s) => s.codec_type === 'audio');
+
+      if (stream) {
+        Object.assign(inspection, {
+          Duration: Math.round(+stream.duration * 1000),
+          Format: stream.codec_name,
+          Bitrate: +stream.bit_rate,
+          Frequency: +stream.sample_rate,
+          Channels: stream.channels,
+          Layout: stream.channel_layout,
+        });
+      }
+    } catch (error) {
+      console.log(error);
+    }
+
+    // Additional inspection for MP3 streams
+    if (inspection.Format === 'mp3') {
+      try {
+        const check = await mpck.inspect(filePath);
+
+        if (check) {
+          Object.assign(inspection, {
```

Comment:
This (and the other cases like it) assume that the properties from the underlying inspection will **all** be available. There may be some value in checking each one individually, to ensure that we're only returning properties with values (i.e., preventing `Layers: null`, if mpck fails to detect the layers of a file).